### PR TITLE
Fix/none dialogue labels

### PR DESCRIPTION
**Pull request by dvilelaf**
## Proposed changes

Update incomplete_to_complete_dialogue_labels so it can contain None values. We can now remove the temporal None in the dialogue label pop and we won't get a key error as keys are automatically added on dialogue creation.

## Fixes

[This](https://github.com/valory-xyz/open-aea/pull/107#discussion_r850955059)

## Types of changes

What types of changes does your code introduce to agents-aea?
_Put an `x` in the boxes that apply_

- [ ] Bugfix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist

_Put an `x` in the boxes that apply._

- [ ] I have read the [CONTRIBUTING](../CONTRIBUTING.md) doc
- [ ] I am making a pull request against the `develop` branch (left side). Also you should start your branch off our `develop`.
- [ ] Lint and unit tests pass locally with my changes and CI passes too
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] I have checked that code coverage does not decrease.
- [ ] I have added necessary documentation (if appropriate)
- [ ] Any dependent changes have been merged and published in downstream modules

## Further comments

If this is a relatively large or complex change, kick off the discussion by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: aea/protocols/dialogue/base.py [hunk 1]
```diff
@@ -1374,14 +1383,15 @@ def _load_terminated_dialogues(self) -> None:
     def _incomplete_dialogues_labels_to_json(self) -> List:
         """Dump incomplete_to_complete_dialogue_labels to json friendly dict."""
         return [
-            [k.json, v.json]
+            [k.json, v.json if v else None]
```
Comment (Adamantios) [inline aea/protocols/dialogue/base.py:1386]: Could use `getattr` with `None` as default

File: aea/protocols/dialogue/base.py [hunk 2]
```diff
@@ -1228,9 +1236,10 @@ def is_dialogue_present(self, dialogue_label: DialogueLabel) -> bool:
 
     def get_latest_label(self, dialogue_label: DialogueLabel) -> DialogueLabel:
         """Get latest label for dialogue."""
-        return self._incomplete_to_complete_dialogue_labels.get(
+        complete_dialogue_label = self._incomplete_to_complete_dialogue_labels.get(
             dialogue_label, dialogue_label
         )
+        return complete_dialogue_label if complete_dialogue_label else dialogue_label
```
[the remark this answers is not in the document]
Comment (dvilelaf) [reply aea/protocols/dialogue/base.py:1242]: Default is dialogue_label for non-existent keys. The problem here is that there are some keys whose value is None. So:
- If key does not exist, return default dialogue_label
- If key exist but value is None, return dialogue_label
- Otherwise return the correspondent complete_dialogue_label

[the remark this answers is not in the document]
Comment (Adamantios) [reply aea/protocols/dialogue/base.py:1242]: Then why add `None` value [here](https://github.com/valory-xyz/open-aea/pull/125/files#diff-c98ed1e8f81fab5ef743ad9a6572e0f077095a0b1cabbf934eed9183ce0ba044R1134-R1144)? Cannot we simply not add a value for the key?

[the remark this answers is not in the document]
Comment (dvilelaf) [reply aea/protocols/dialogue/base.py:1242]: Yes, before we hadn't those dummy keys, but quoting David M: "_by design all incomplete dialogue labels must be present in this mapping_". The existence of those keys is checked in several places so I'm adding _key: None_ for all keys that are going to be checked. Not sure if it's a proper solution but it is the only one I found that fixed all the failing tests.

Comment (DavidMinarsch) [inline aea/protocols/dialogue/base.py:1242]:
```suggestion
        return self._incomplete_to_complete_dialogue_labels.get(dialogue_label) or dialogue_label
```
we know key exists and if value is None we get back dialogue_label

File: aea/protocols/dialogue/base.py [hunk 3]
```diff
@@ -1855,9 +1865,7 @@ def _complete_dialogue_reference(self, message: Message) -> None:
             incomplete_dialogue_reference, message.sender, self.self_address,
         )
 
-        if self._dialogues_storage.is_dialogue_present(
```
Comment (DavidMinarsch) [inline aea/protocols/dialogue/base.py:1858]: why the change?

Comment (dvilelaf) [reply aea/protocols/dialogue/base.py:1858]: Now we have this _{incomplete_label: None}_ entries, _incomplete_dialogue_label_ will always be in the incomplete mapping, so if we keep this conditional we would never enter it.

**Review by DavidMinarsch — APPROVED** [review]
LGTM
